src/model/transaction/AccountAddressRestrictionTransaction.ts

```diff
@@ -190,4 +195,59 @@ export class AccountAddressRestrictionTransaction extends Transaction {
         );
         return transactionBuilder.serialize();
     }
+
+    /**
+     * @internal
+     * @param receiptHttp ReceiptHttp
+     * @param aggregateTransactionIndex Transaction index for aggregated transaction
+     * @returns {Observable<AccountAddressRestrictionTransaction>}
+     */
+    resolveAliases(receiptHttp: ReceiptHttp, aggregateTransactionIndex?: number): Observable<AccountAddressRestrictionTransaction> {
```

Comment:
I am not fan of the double dispatch. It gets the Transaction model too complicated with internal methods the user doesn't really care. I would create a switch/handers statement in the TransactionService (especially when there is a static call  TransactionService.getResolvedFromReceipt) and remove resolveAliases from the transaction hierarchy.

I would stop using the Http implementations and start using the Repository Interface, in this case, ReceiptRepository

Reply:
Do you mean to put all codes in to one class here? It's another option agree but the single class  can be complicated and main thing here is you will lose the single responsibility for Transaction to resolve alias. 

Regrading the interface, ts doesn't work like JAVA or C# which you can use unity or spring, There's no DI in TS SDK, so  resolving the interface is not easy

Reply:
Not about DI, it's just about using interfaces, you can do

`resolveAliases(recipientRepository: RecipientRepository, aggregateTransactionIndex?: number)`

Reply:
How do you find the correct implementation of the injected interface?